src/components/Pressable/Pressable.tsx

```diff
@@ -0,0 +1,269 @@
+
+  const touchGesture = useMemo(
+    () =>
+      Gesture.Manual()
+        .onTouchesDown((event) => {
+          handlingOnTouchesDown.current = true;
+          pressableRef.current?.measure((_x, _y, width, height) => {
+            // pressableRef.current?._nativeTag yields targetId on android & ios
+            if (
+              !isTouchWithinInset(
+                {
+                  width,
+                  height,
+                },
+                normalizedHitSlop,
+                event.changedTouches.at(-1)
+              ) ||
+              isPressedDown.current
+            ) {
+              return;
+            }
+
+            isPressedDown.current = true;
+
+            if (props.unstable_pressDelay) {
+              pressDelayTimeoutRef.current = setTimeout(() => {
+                pressInHandler(event);
+              }, props.unstable_pressDelay);
+            } else {
+              pressInHandler(event);
+            }
+
+            onEndHandlingTouchesDown.current();
+            handlingOnTouchesDown.current = false;
+          });
+        })
+        .onTouchesUp((event) => {
+          if (handlingOnTouchesDown.current) {
+            onEndHandlingTouchesDown.current = () => pressOutHandler(event);
+            return;
+          }
+          pressOutHandler(event);
+        })
+        .onTouchesCancelled((event) => {
+          if (
+            !isPressedDown.current ||
+            event.allTouches.length > event.changedTouches.length
+          ) {
+            return;
+          }
+
+          // original triggers onPressOut on cancel, but not onPress
+          props.onPressOut?.(adaptTouchEvent(event));
+
+          isPressedDown.current = false;
+          setPressedState(false);
+        }),
+    [
+      props.onPress,
+      props.onPressIn,
+      props.onPressOut,
+      setPressedState,
+      isPressedDown,
+      isPressCallbackEnabled,
+      normalizedHitSlop,
+      pressDelayTimeoutRef,
+    ]
+  );
+
+  // rippleGesture lives inside RNButton to enable android's ripple
+  const rippleGesture = useMemo(() => Gesture.Native(), []);
+
+  pressGesture.minDuration(
+    (props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION) +
+      (props.unstable_pressDelay ?? 0)
+  );
+
+  const appliedHitSlop = addInsets(
+    normalizedHitSlop,
+    normalizedPressRetentionOffset
+  );
+
+  const isPressableEnabled = props.disabled !== true;
+
+  touchGesture.enabled(isPressableEnabled);
+  pressGesture.enabled(isPressableEnabled);
+  hoverGesture.enabled(isPressableEnabled);
+  rippleGesture.enabled(isPressableEnabled);
+
+  touchGesture.shouldCancelWhenOutside(true);
+  pressGesture.shouldCancelWhenOutside(true);
+  hoverGesture.shouldCancelWhenOutside(true);
+  rippleGesture.shouldCancelWhenOutside(true);
+
+  if (Platform.OS === 'web') {
+    touchGesture.shouldCancelWhenOutside(false);
+    pressGesture.shouldCancelWhenOutside(false);
+    hoverGesture.shouldCancelWhenOutside(false);
+  }
```

Comment:
> While it is default on browsers, do we want it to be?

Android also [does this by default](https://github.com/software-mansion/react-native-gesture-handler/blob/31a62e96a006ec655175aa8259160415e0f96988/android/src/main/java/com/swmansion/gesturehandler/core/GestureHandler.kt#L64). Same for [iOS](https://github.com/software-mansion/react-native-gesture-handler/blob/31a62e96a006ec655175aa8259160415e0f96988/apple/RNGestureHandler.m#L101). The thing is, when it is not set to `false`, handlers will cancel when you move outside of the underlying view. It means that, for example, if user will be panning fast, gesture may be cancelled if he leaves view's boundaries. We don't want that behavior 😅

Reply:
Ok but as far as I see, the original `Pressable` has `shouldCancelWhenOutside` set to `true`, and we want the new `Pressable` to behave like the old one. I'm genuinely not sure if I'm missing something here.

https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/0771cdf5-6ada-4068-99f1-fc92fc1b918c

Reply:
And to be clear, removing the `shouldCancelWhenOutside` block as you requested, so that the default value is used looks like this:


https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/5d26ab46-c614-43ad-b6aa-d9cbe2e29e16

Reply:
The videos are captured on `iOS` and its implementation was fine. I was talking about web behavior. If you leave `shouldCancelWhenOutside` set to `false` on web, both `Pressables` act in the same way and this is what we want to achieve.

Of course I may be wrong, so If you see that something is not right with this approach feel free to tell me 😅 


https://github.com/software-mansion/react-native-gesture-handler/assets/63123542/2d918e7a-d920-47d8-823a-3b4e94d0a250

Reply:
Alight i now understand what you meant all this time. I believe this feature is implemented as of now, sorry for the fuss 😄